### PR TITLE
fabtests: Utilize `fi_setopt`/`fi_getopt` sequence for inject size (`-j`), prov/efa: Update EP's `inject_size` in zero-copy mode

### PR DESCRIPTION
**Pull request by darrylabbate**
This also:
- Calls `fi_setopt(..., FI_OPT_INJECT_{MSG,RMA}_SIZE)` if `-j` is specified
- Retrieves the inject size via `fi_getopt()` in the pingpong/bw fabtests functions; falling back to `fi->tx_attr->inject_size` unless `-j` was specified
- Adds support for `FI_OPT_INJECT_RMA_SIZE` in the EFA provider, which was overlooked originally

### CODE REVIEW

**Review by shijin-aws — COMMENTED** [review]
Changes LGTM, can you add unit test as part of the commit

**Comment by darrylabbate** [comment]
ack

**Comment by shijin-aws** [comment]
AWS CI failure shows you need some changes to fabtests

**Comment by shijin-aws** [comment]
Need to onboard fabtests to the model that: try to query the inject size from fi_getopt, if it's not supported, fallback to the inject size in fi_info

**Comment by a-szegel** [comment]
bot:aws:retest

File: fabtests/benchmarks/benchmark_shared.c [hunk 1]
```diff
@@ -145,10 +153,17 @@ int pingpong(void)
 
 int pingpong_rma(enum ft_rma_opcodes rma_op, struct fi_rma_iov *remote)
 {
-	int ret, i, inject_size;
+	int ret, i;
+	size_t inject_size = fi->tx_attr->inject_size;
+
+	ret = fi_getopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_INJECT_RMA_SIZE, &inject_size, &(size_t){sizeof inject_size});
+	if (ret && ret != -FI_ENOPROTOOPT) {
+		FT_PRINTERR("fi_getopt(FI_OPT_INJECT_MSG_SIZE)", ret);
```
Comment (shijin-aws) [inline fabtests/benchmarks/benchmark_shared.c:161]: `FI_OPT_INJECT_MSG_SIZE` -> `FI_OPT_INJECT_RMA_SIZE`

**Review by shijin-aws — COMMENTED** [review]
unit test for efa provider change is still needed.

**Comment by shijin-aws** [comment]
Can you modify the PR overview to reflect the fabtests change

File: fabtests/benchmarks/benchmark_shared.c (hunk 1 above)
Comment (darrylabbate) [reply fabtests/benchmarks/benchmark_shared.c]: Done

**Comment by darrylabbate** [comment]
> unit test for efa provider change is still needed.

Added some `assert_*`s to ensure the correct inject size after enabling the EP in the zero-copy tests.

**Review by shijin-aws — APPROVED** [review]
LGTM, thx

**Comment by j-xiong** [comment]
Very long lines for the fi_setopt() and fi_getopt() calls in the first commit. Could them be broken down?

**Comment by darrylabbate** [comment]
> Very long lines for the fi_setopt() and fi_getopt() calls in the first commit. Could them be broken down?

ack

**Review by j-xiong — APPROVED** [review]

**Comment by shijin-aws** [comment]
Is Intel CI failure related?

**Comment by j-xiong** [comment]
@shijin-aws There are two types of failues: (1) sockets provider failed some shared context tests, that is most likely unrelated. (2) ucx provider failed some tests due to returning the wrong error code for unsupported options. That's the issue with the ucx provider, not this PR.

**Comment by shijin-aws** [comment]
@j-xiong Thanks

**Comment by shijin-aws** [comment]
@darrylabbate can you backport to v1.22.x

**Comment by darrylabbate** [comment]
ack

**Pull request merged**